integration-tests/src/test/java/oracle/kubernetes/operator/BaseTest.java

```diff
@@ -520,4 +584,41 @@ private void callWebAppAndVerifyScaling(Domain domain, int replicas) throws Exce
       TestUtils.checkServiceCreated(podName, domainNS);
     }
   }
+
+  public static void tearDown() throws Exception {
+    logger.log(
+        Level.INFO,
+        "TEARDOWN: Starting Test Run TearDown (cleanup and state-dump)."
+            + " Note that if the test failed previous to tearDown, "
+            + " the error that caused the test failure may be reported "
+            + "after the tearDown completes. Note that tearDown itself may report errors,"
+            + " but this won't affect the outcome of the test results.");
+    StringBuffer cmd =
+        new StringBuffer("export RESULT_ROOT=$RESULT_ROOT && export PV_ROOT=$PV_ROOT && ");
+    cmd.append(BaseTest.getProjectRoot())
+        .append("/integration-tests/src/test/resources/statedump.sh");
+    logger.info("Running " + cmd);
+
+    // renew lease before callin statedump.sh
+    TestUtils.renewK8sClusterLease(getProjectRoot(), getLeaseId());
```

Comment:
If this renew (or _any_ renew) fails, we must exit immediately -- all future results , state dumps, and cleanup are invalid.

Reply:
Yes, it exits with Exception.

Reply:
Cool - resolved!